### ISSUE TITLE
Arduino Leonardo clone (Pro Micro) fails to reset and upload from Raspberry Pi

### ISSUE THREAD

**Issue by paulr4** [amperka/ino#271]
I'm trying to program an Arduino Pro Micro (Leonardo based) from a Raspberry Pi. I've taken the following steps but haven't yet succeeded in uploading anything using ino, or anything else.
- Installed arduino-core (It's a headless setup so I'm assuming this is all I need)
- Installed ino using `sudo pip install ino`
- Followed the quickstart tutorial [here](http://inotool.org/quickstart). (In the code I changed the LED pin number to 17, as is needed on a Pro Micro)
- Built the project using `sudo ino build -m leonardo` (successfully)
- Attempted to upload using `sudo ino upload -m leonardo`
  It returns:
  `Guessing serial port ... /dev/ttyACM0` (which is correct)
  And then stops. I have to Ctrl-C to get out of it.

I've read about similar problems but no solution I've found has worked for me. Writing my own Python script to connect and disconnect at 1200 baud does not reset my Arduino.
I've tried [this patch](https://github.com/bendenoz/ino/commit/d9b7c281d71bc981a75bbd4fef5297ad193189a2#diff-5ed356d4542c24c6ce0440f8fe43d840R111), as suggested on another thread with a similar problem. No change.
The same Arduino resets as expected when programming from Windows, using the Leonardo settings, so I'm confident the hardware is not at fault.

**Comment by paulr4** [comment]
I discovered two problems:

A process running on my machine which was blocking access to the port and causing the crash, this is fixed now, so upload.py runs to completion.

The patch linked above uses the deprecated `setBaudrate(1200)` command, so I've changed it to `ser.baudrate = 1200` instead.

It's still not working though. The code runs, waits for the reset, then attempts to run AVRDude, but the Arduino has not been reset into bootloader mode, so it can't connect, and returns:

`Connecting to programmer: .avrdude: butterfly_recv(): programmer is not responding`
